backend/tests/integration/common_utils/managers/user.py

```diff
@@ -101,41 +113,118 @@ def verify_role(user_to_verify: DATestUser, target_role: UserRole) -> bool:
     def set_role(
         user_to_set: DATestUser,
         target_role: UserRole,
-        user_to_perform_action: DATestUser | None = None,
-    ) -> None:
-        if user_to_perform_action is None:
-            user_to_perform_action = user_to_set
+        user_performing_action: DATestUser | None = None,
+    ) -> DATestUser:
```

Comment:
In OOP context setter methods typically modify objects in place and do not return something separate. However, we are trying to be more functional. So prefer returning a separate object with the new state, but it's very important then to not also modify the state of the existing object. That becomes very unexpected for the caller.